### ISSUE TITLE
Colliders not updating when quad is rotated

### ISSUE THREAD

**Issue by system32uwu** [ceramic-engine/ceramic#119]
Repro: https://github.com/system32uwu/ceramic-flappy-bird

added some visual debugging by drawing the body shapes on collision, pretty sure it has to do with rotation because if i set it to 0 then it has no issues

looking here but i'm not familiar with the codebase just yet: https://github.com/ceramic-engine/ceramic/blob/master/plugins/arcade/runtime/src/ceramic/VisualArcadePhysics.hx

**Comment by system32uwu** [comment]
It's actually related to the anchor, just fixed everything by setting all quads to anchor(0.5,05)

may be what you explain here in this comment: https://github.com/jeremyfa/arcade/blob/master/arcade/Body.hx#LL89C1-L94C35
probably worth adding to the documentation

**Comment by jeremyfa** [comment]
Yes, the rotation is quite rudimentary (on purpose) when using arcade physics.

Beware that the api docs in the arcade repository are far from being perfect because they are ported from Phaser 2 arcade physics, and could mention things (like "sprite") that are not really relevant.

**Comment by system32uwu** [comment]
Sure, no problem, it's just different from some other physics engine.

I'm currently trying to achieve collisions with something like this (sword rotating around the player, like in survivor-like games):

```
import ceramic.Assets;
import ceramic.Quad;
import ceramic.SpriteSheet;

class Sword extends TaggedSprite {
	var scene:MainScene;
	var collider:Quad;

	public function new(assets:Assets, scene:MainScene) {
		super([Tags.PlayerWeapon]);
		this.scene = scene;

		sheet = new SpriteSheet();
		sheet.texture = assets.texture(Images.PIRATE_MAIN_WEAPON);
		sheet.grid(9, 31);
		sheet.addGridAnimation('default', [0], 0);
		animation = 'default';

		initArcadePhysics();

		anchor(0, 2);

		scale(2);

		gravity(0, 0);

		this.onCollide(this, (v1, v2) -> {
			trace('collide sword');
		});
	}

	override function update(dt:Float) {
		rotation += 360 * dt / 3;
		super.update(dt);
	}
}
```

So if you draw a quad here in the update method for the position of this sword, it'll always be in 0,0, even if it's attached to the player and the player starts moving, so yeah it's a little confusing... Also tried attaching a child quad to the sword and make that handle the collisions instead, but now the problem is that the collider will always stay straight, which is visually weird as the enemies get hit beforehand (sword is slightly rotated so it shouldn't just yet hit the target...).

```
import ceramic.Assets;
import ceramic.Color;
import ceramic.Quad;
import ceramic.SpriteSheet;

class Sword extends TaggedSprite {
	var scene:MainScene;
	var collider:Quad;

	public function new(assets:Assets, scene:MainScene) {
		super([Tags.PlayerWeapon]);
		this.scene = scene;

		sheet = new SpriteSheet();
		sheet.texture = assets.texture(Images.PIRATE_MAIN_WEAPON);
		sheet.grid(9, 31);
		sheet.addGridAnimation('default', [0], 0);
		animation = 'default';

		initArcadePhysics();

		anchor(0, 2);

		scale(2);

		gravity(0, 0);

		collider = new Quad();
		collider.x = 0;
		collider.y = 0;
		collider.width = width;
		collider.height = height;
		collider.color = Color.RED;
		collider.anchor(0, 2);
		this.collider.allowRotation = true;

		this.collider.onCollide(this, (v1, v2) -> {
			trace('collide sword');
			trace('--');
		});
	}

	override function update(dt:Float) {
		rotation += 360 * dt / 3;

		// based on the rotation, calculate where the collider should be
		var angle = rotation * Math.PI / 180;
		var _x = Math.cos(angle) * 64;
		var _y = Math.sin(angle) * 64;
		this.collider.x = parent.x + _x;
		this.collider.y = parent.y + _y;
		this.collider.rotation = rotation;

		app.arcade.world.collide(this.collider, scene.enemies);

		super.update(dt);
	}
}
```
I've just seen you're using nape for your ld project: https://github.com/jeremyfa/ld49 in which there are some rotations there, shall I use that instead?

**Comment by system32uwu** [comment]
Trying something here: https://github.com/system32uwu/ceramic-survivors-like/blob/nape/src/MainScene.hx

- The collision between the sword (child of the player) is not working
- The callbacks never get fired

Not sure what I'm doing wrong there, collision itself between the enemy and the player works, but doesn't fire the callback, regardless of the interaction type, and for the sword the only difference is that it's a child of the player, but I don't know why that disables it from colliding with other objects

**Comment by jeremyfa** [comment]
You should keep in mind that the current implementation of physics in Ceramic doesn't take into account visual parents (mostly for performance reasons). In a way, it's like all physics bodies are in the same container.

I'd like to make it an option eventually to be able to compute a position (and size) relative to the parent, but that's not my priority right now.

Anyway, you can still overcome this limitation by using `arcade.onUpdate()` event. From there, you make custom edits of your visual's bodies (like adding parent visual position explicitly to a child's `body.x` and `body.y`) before you test collisions. You can also create your own `arcade.Body` objects that you set how you want and collide with them.

And yes, nape physics can handle more advanced collisions (but still no visual parent handling). The setup can be a bit more complicated too, and more expensive to compute. Depends on your use case.

**Comment by system32uwu** [comment]
hey, in case anyone has the same issue, i fixed it manually setting the body rotation to the sprite rotation, plus what jeremy suggested of having something bound to the position of the player,

https://github.com/system32uwu/ceramic-survivors-like/commit/9c087f2a69fb1aa00505be5d352f5ff43c36dafc

See RotatorCenter.hx and Sword.hx
